### ISSUE TITLE
Support ARFs with `oscap xccdf generate fix`

### ISSUE THREAD

**Issue by mpreisler** [OpenSCAP/openscap#339]
Wow, I had no idea we don't support ARFs there.

```
$ oscap xccdf generate fix arf.xml 
OpenSCAP Error: Session input file was determined but it isn't an XCCDF file, a source datastream or an XCCDF tailoring file. [xccdf_session.c:135]
```

Why is `generate fix` expecting XCCDF file, SDS or tailoring file? Shouldn't it expect result files such as XCCDF results or ARFs? Or, if it generates fixes assuming all rules failed, shouldn't it allow result files as well as benchmarks?

**Comment by jan-cerny** [comment]
Is this really a certification blocker?

**Comment by mpreisler** [comment]
Probably not but it is a usability issue for sure.

**Comment by ybznek** [comment]
https://github.com/OpenSCAP/openscap/issues/524 is related issue .

As far as I know, we could have two ways how to generate fix if we used tailoring file:
 1) use arf results (but we don't support it)
 2) specify xccdf result file and used tailoring file (but user cannot specify used tailoring file)

It caused me problems with writing of ansible blogpost and I think we should fix it early.

I rather to have fixed https://github.com/OpenSCAP/openscap/issues/339 than https://github.com/OpenSCAP/openscap/issues/525.

**Comment by rsprudencio** [comment]
Merged at #659
